src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,83 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.Scanner;
+
+public class Storage {
+    protected String filePath;
+    protected ArrayList<Task> fileData;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.fileData = new ArrayList<Task>();
+    }
+
+    /**
+     * Returns an ArrayList<Task> based on data represented in the Duke.txt file.
+     * If the file does not exist, it creates a new directory and new file.
+     *
+     * @param filePath File Path for Duke.txt
+     * @return ArrayList<Task> based on data parsed from the Duke.txt
+     * @throws DukeException If Deadline object is not instantiated properly
+     * @throws IOException If file is not located or not available
+     */
+    public ArrayList<Task> load(String filePath) throws DukeException, IOException {
```

Comment:
Code looks good to me. However, it could be further improved by making Task an abstract class, so you can create methods for each case.